### DISCUSSION TITLE
How to prevent pyInstaller temporary files from being deleted?

### DISCUSSION THREAD

**Discussion by ugurcanakyuz** [pyinstaller/pyinstaller#6765] [PyInstaller] [answered]
I am creating an executable of Flask app by using pyinstaller. It allocates 10GB storage in the _MEIPASS folder under Temp directory. However, temporary files in _MEIPASS folder are removed when the program is terminated. It tooks long time to load all the files each startup. How can I prevent the pyinstaller temporary files from being deleted?

I want the exe created by pyinstaller to be uncompressed only once into specific folder and not being deleted when the exe is terminated. Is it possible?

**Comment by bwoodsend** [comment, marked as answer]
It's deliberately not supported. If you want a onefile single unpack then use something like [inno setup](https://jrsoftware.org/isinfo.php) to create an installer from a onedir build. But irregardless of what you do, you shouldn't be writing to `_MEIPASS` - it's a good security practice to install applications into directories requiring admin privileges to modify and by writing into an application at runtime you block this. Find the proper place to put these files.

**Comment by ugurcanakyuz** [comment]
Thanks for your answer. I will try to create a setup.
